lib/src/codegen/js_codegen.dart

```diff
@@ -1025,20 +1026,25 @@ class JSCodegenVisitor extends GeneralizingAstVisitor with ConversionVisitor {
       /// Rename private names so they don't shadow the private field symbol.
       /// The renamer would handle this, but it would prefer to rename the
       /// temporary used for the private symbol. Instead rename the parameter.
-      return new JSTemporary('${name.substring(1)}');
+      return _getTemp(e, '${name.substring(1)}');
     }
 
     if (_isTemporary(e)) {
       if (name[0] == '#') {
         return new JS.InterpolatedExpression(name.substring(1));
       } else {
-        return new JSTemporary(name);
+        return _getTemp(e, name);
       }
     }
 
     return new JS.Identifier(name);
   }
 
+  JSTemporary _getTemp(Object key, String name) {
+    var t = _temps[key];
+    return t != null ? t : _temps[key] = new JSTemporary(name);
```

Comment:
JSTemporary _getTemp(Object key, String name) => _temps.putIfAbsent(key, () => new JsTemporary(name);

also, even if that method existed, doing an assignment in a clause of a conditional is scary and you should likely use if () { } else {} instead for increased readability.